testing/sdk_auth.py

```diff
@@ -0,0 +1,376 @@
+    rc, msg = sdk_marathon.install_app(app_definition)
+    if not rc:
+        raise RuntimeError("Can't install KDC marathon app: {err}".format(err=msg))
+
+
+@retry(stop_max_attempt_number=3, wait_fixed=2000)
+def _check_kdc_marathon_task_is_running() -> bool:
+    """
+    :return (bool): Indicator of whether the KDC marathon app is running or not. Raises exception if app is not detected
+    after the specified wait & retry period.
+    """
+    log.info("Waiting for app to be running...")
+    shakedown.wait_for_task("marathon", KERBEROS_APP_ID)
+
+
+def _get_task_id() -> str:
+    """
+    :return (str): The ID of the task running the KDC app.
+    """
+    log.info("Getting task id")
+    raw_tasks = sdk_cmd.run_cli("task --json")
+    if raw_tasks:
+        tasks = json.loads(raw_tasks)
+        for task in tasks:
+            if task["name"] == KERBEROS_APP_ID:
+                log.info("Task id is {}".format(task["id"]))
+                return task["id"]
+
+    raise RuntimeError("Expecting marathon KDC task but no such task found. Running tasks: {tasks}".format(
+        tasks=raw_tasks))
+
+
+def _get_framework_id() -> str:
+    """
+    :return (str): The ID of the framework running the KDC app.
+    """
+    log.info("Getting framework id")
+    raw_tasks = sdk_cmd.run_cli("task --json")
+    if raw_tasks:
+        tasks = json.loads(raw_tasks)
+        for task in tasks:
+            if task["name"] == KERBEROS_APP_ID:
+                log.info("Framework id is {}".format(task["framework_id"]))
+                return task["framework_id"]
+
+    raise RuntimeError("Expecting marathon KDC task but no such task found. Running tasks: {tasks}".format(
+        tasks=raw_tasks))
+
+
+@retry(stop_max_attempt_number=2, wait_fixed=1000)
+def _get_host_id() -> str:
+    """
+    :return (str): The ID of the node running the KDC app.
+    """
+    log.info("Getting host id")
+    raw_tasks = sdk_cmd.run_cli("task --json")
+    if raw_tasks:
+        tasks = json.loads(raw_tasks)
+        for task in tasks:
+            if task["name"] == KERBEROS_APP_ID:
+                log.info("Host id is {host_id}".format(host_id=task["slave_id"]))
+                return task["slave_id"]
+
+    raise RuntimeError("Expecting marathon KDC task but no such task found. Running tasks: {tasks}".format(
+                        tasks=raw_tasks))
+
+
+@retry(stop_max_attempt_number=2, wait_fixed=1000)
+def _get_host_name(host_id: str) -> str:
+    """
+    Fetches the host name for the host running the KDC app.
+    :param host_id (str): The ID of the host, used to look up the appropriate node.
+    :return (str): Name of the host running the KDC app.
+    """
+    log.info("Getting host name")
+    raw_nodes = sdk_cmd.run_cli("node --json")
+    if raw_nodes:
+        nodes = json.loads(raw_nodes)
+        for node in nodes:
+            if node["id"] == host_id:
+                log.info("Host name is {host_name}".format(host_name=node["hostname"]))
```

Comment:
Same here (see comment w.r.t framework id and task id).

Reply:
Not quite the same here, as we're going through the nodes and not the tasks.